### PR TITLE
Add '--hostfile' option for MPI engine.

### PR DESCRIPTION
**Pull request by lyndonhenry**
This PR adds a new `--hostfile` option for Souffle when compiled with `--enable-mpi`. The file given as the value of this option will be passed to the `--hostfile` option in the call to `mpiexec`. We do this because specifying a hostfile allows MPI processes to be allocated across multiple cores of multiple machines. For more information, as OpenMPI is the target implementation of MPI for Souffle, refer to [this link right here](https://www.open-mpi.org/faq/?category=running#mpirun-hostfile).

### CODE REVIEW

**Comment by souffle-lang** [comment]
[64% (+0.02%) vs master 64%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/110/)

**Comment by mmcgr** [comment]
Jenkins, test this please

**Comment by souffle-lang** [comment]
[64% (+0.01%) vs master 64%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/111/)

**Comment by souffle-lang** [comment]
[64% (0.0%) vs master 64%](http://plang1.it.usyd.edu.au/jenkins/job/PR-RHEL7-gcc7-Coverage/112/)

File: src/main.cpp [hunk 1]
```diff
@@ -82,7 +82,14 @@ void executeBinary(const std::string& binaryFilename
     int exitCode;
 #ifdef USE_MPI
     if (Global::config().get("engine") == "mpi") {
-        exitCode = system(("mpiexec -n " + std::to_string(numberOfProcesses) + " " + binaryFilename).c_str());
+        std::stringstream ss;
```
Comment (mmcgr) [inline src/main.cpp]: Could you change this so the mpi option exists even if mpi is disabled? i.e. give an error message, or a warning, if mpi is disabled and the option is used.

Comment (lyndonhenry) [reply src/main.cpp]: This has been addressed in a follow up at #694.

**Pull request merged**